server/lib/middleware.js

```diff
@@ -79,7 +82,22 @@ module.exports = function middlewareConstructor(config) {
 
       // Decrypt oauth token
       req.user = req.session.user;
-      req.user.token = cryptr.decrypt(req.session.token);
+      let token = req.user.token = cryptr.decrypt(req.session.token);
+
+      if(ASSERT_TOKEN) {
+        // We expect a token of the form '16d4c4d2fa4d6e4aa6b1b5c6a115ad44fd271dd98204f2008bf5efbba5a56dec'
```

Comment:
can we change this to "we expect an X character long, hexadecimal character string"? And directly related to that, if we know the exact length we expect, we can assert on token length as well